### ISSUE TITLE
WebGL / GPU issues

### ISSUE THREAD

**Issue by stevelodin** [cloudtracer/ThreatPinchLookup#72]
Switched to a new endpoint (AWS t2.xlarge, Citrix, Win10 VDI) that doesn't render the settings pages so I can add a new API search tool.  Confirmed it wasn't AV or Secure Web Gateway causing problems.  Chrome developer tool shows:

THREE.WebGLRenderer 90
app.js:7 THREE.WebGLRenderer: Error creating WebGL context.
xt @ app.js:7
app.js:7 Uncaught TypeError: Cannot read property 'getExtension' of null
    at Object.get (app.js:7)
    at o (app.js:7)
    at new xt (app.js:7)
    at stateInit (app.js:1)
    at app.js:3
    at Object.693../SpriteText.js (app.js:48)
    at o (app.js:1)
    at app.js:1
    at Object.706.../components/ForcedGraph.js (app.js:51)
    at o (app.js:1)

Doing some sleuthing shows this is possibly a GPU-related topic with WebGL.  I've tried all the suggested Chrome (settings, flags) to see if it fixed it, but no success.

Chrome GPU shows:

WebGL: Hardware accelerated but at reduced performance
WebGL2: Hardware accelerated but at reduced performance

But WebGL browser settings shows:

Browser User-Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36

This browser supports WebGL | ×False (supported, but disabled in browser settings, or blocked by extensions)
This browserr supports WebGL 2 | ×False (supported, but disabled in browser settings, or blocked by extensions)

I'm trying to see what my options are (rather than getting a new physical laptop).  Anyone heard of this related to ThreatPinch and WebGL and AWS low-end GPU emulation?

Thanks,
Steve

**Comment by cloudtracer** [comment]
Hi @stevelodin,

Thanks for the detailed report, unfortunately I don't have a quick answer for you but since you're having the issue in an AWS instance I should be able to spin it up and reproduce over the weekend.

I'm not sure if this will help your situation or not, but in the mean time I did leave the old plain-ish HTML version of the options page in the extension in case of weird JS issues or for debugging which you can access with this URL: chrome-extension://ljdgplocfnmnofbhpkjclbefmjoikgke/src/options/options-developer.html

**Comment by stevelodin** [comment]
Got it.  That works in the meantime.  Thanks.

Webglreport.com shows:

_× This browser supports WebGL 2, but it is disabled or unavailable.

Sometimes this is the result of older video drivers being rejected by the browser. Try updating your video drivers if possible.

Also check out Get WebGL, or try installing the latest version of Chrome, or Firefox._


My desktop guru tried to see if he could update the video drivers but no quick success there.

Thanks,
Steve

**Comment by cloudtracer** [comment]
@stevelodin - I just pushed out an update in 3.0.6, should be available in the next hour or so. I haven't managed to replicate the problem yet, but I've added some code which will try to determine if the browser is WebGL capable, and if not, it won't render the graph. The graph will just be replaced with message indicating that the browser isn't capable of rendering it.

I might try to get fancy with this in the future and revert to an older version of the graph, but my time limited right now.

Let me know if 3.0.6 fixes your issue, if it doesn't I'll spend the extra time to get it properly reproduced.
